### PR TITLE
feat: auto refresh az app config background job

### LINKED ISSUE #80: Provide background job for asynchronous configuration changes in Azure App Configuration
**Is your feature request related to a problem? Please describe.**
Provide background job for asynchronous configuration changes in Azure App Configuration so that instances can refresh the values

**Describe the solution you'd like**
An approach similar to Azure Key Vault invalidation.

**Describe alternatives you've considered**
N/A

**Additional context**
@gverstraete can help here.

Reload data from Azure App Configuration: https://docs.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-dotnet-core#reload-data-from-app-configuration

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds a new background job that refreshed the `IConfiguration` within a project, based on consumed events from the Azure App Configuration resource.

Closes #80

### CODE REVIEW

File: docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+## Installation
+To use these features, you have to install the following package:
+
+```shell
+PM > Install-Package Arcus.BackgroundJobs.AzureAppConfiguration
+```
+
+## Setup
+To automtically notify your `IConfiguration` instance in your application for any change in the remote Azure App Configuration, you need some setup.
```
Comment (fgheysels) [inline docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md:17]:
```suggestion
To automatically notify your `IConfiguration` instance in your application for any change in the remote Azure App Configuration, you need some setup.
```

File: docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+We use the events from Azure App Configuration which will be send towards an Azure Service Bus Topic. The background job will look for those events on the topic, and will in turn refresh the `IConfiguration`.
```
Comment (fgheysels) [inline docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md:18]: Is this done via EventGrid ?

Comment (fgheysels) [reply docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md:18]: Might be nice if we have a graphical overview of how this works, and how the different components work together ?

Comment (stijnmoreels) [reply docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md:18]: Yeah, we could do that. But I don't know where the Azure icons are, that we can use for this. 😅 The official package doesn't include Azure App Configuration.

Comment (stijnmoreels) [reply docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md:18]: We could probably also benefit from a ARM template to generate this setup?

Comment (stijnmoreels) [reply docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md:18]: Added an overview pic, https://github.com/arcus-azure/arcus.backgroundjobs/pull/84/commits/d3a7b50560f05ef020cfebb555d04eb8aa618318. Made sure it has a white background so it's still visible in a night-themed version of the site.

Comment (fgheysels) [reply docs/preview/02-Features/05-AzureAppConfiguration/auto-refresh-app-configuration.md:18]: Great thanks!

**Review by fgheysels — APPROVED** [review]

**Pull request merged**